### ISSUE TITLE
require-jsdoc / require-props-default-doc: doc comment detection is unreliable

### ISSUE THREAD

**Issue by ren-yamanashi** [ren-yamanashi/eslint-cdk-plugin#543]
### Describe the bug

The documentation checks in `require-props-default-doc` and `require-jsdoc` accept comments they should not, in three independent ways:

**1. `@default` is matched as a substring.** `comment.value.includes("@default")` matches any occurrence, e.g. inside an e-mail address or a word like `@defaultValue`, so a property with no `@default` tag passes.

```ts
export interface SubstringProps {
  /** Contact no-reply@defaultmail.com if needed */
  readonly bugCase?: string;     // passes require-props-default-doc without a @default tag
  readonly controlCase?: string; // control: reported
}
```

**2. Non-JSDoc block comments satisfy `require-props-default-doc` but not `require-jsdoc`.** The `@default` check only requires `value.includes("*")`, while `require-jsdoc` requires `value.startsWith("*")` (a real `/** ... */`). The same comment therefore passes one rule and fails the other:

```ts
export interface ConsistencyProps {
  /*
   * @default undefined
   */
  readonly sample?: string; // require-props-default-doc: pass / require-jsdoc: missingJSDoc
}
```

**3. A trailing comment on the previous line is credited to the next property.** Both rules use `sourceCode.getCommentsBefore(node)`, which returns any comment between the previous token and the node — including a comment on the previous property's line:

```ts
export interface TrailingProps {
  /** @default 1 */
  readonly a?: number; /** @default 2 */
  readonly b?: number;   // not reported: a's trailing comment counts as b's doc
  readonly control?: number; // control: reported
}
```

### Expected Behavior

- 1: `bugCase` reported — `@default` should be detected as a JSDoc tag, not a substring.
- 2: both rules should agree on what counts as a JSDoc comment (`/** ... */`), so the property is treated consistently.
- 3: `b` reported — a trailing comment that documents `a`'s line should not satisfy `b`.

### Current Behavior

Only the controls are reported:

```
repro/require-props-default-doc/substring.ts        4:3  error  Optional property 'controlCase' in Props interface must have @default JSDoc documentation  awscdk/require-props-default-doc
repro/require-props-default-doc/trailing-comment.ts 5:3  error  Optional property 'control' in Props interface must have @default JSDoc documentation      awscdk/require-props-default-doc
repro/jsdoc-consistency/block-comment.ts            5:3  error  Property 'sample' should have a JSDoc comment  awscdk/require-jsdoc
```

Oxlint reports the identical set.

### Plugin

eslint-plugin-awscdk, oxlint-plugin-awscdk

### Plugin Version

eslint-plugin-awscdk 4.4.0, oxlint-plugin-awscdk 1.0.0

### Language Version

TypeScript 6.0.3 (eslint-plugin-awscdk) / TypeScript 7 preview via oxlint-tsgolint (oxlint-plugin-awscdk)

### Additional Context

- Verified on `main` (35b889a), rule-isolated configs.
- Sources (identical in both packages): `packages/*/src/rules/require-props-default-doc.ts` (`includes("*") && includes("@default")`), `packages/*/src/rules/require-jsdoc.ts` (`startsWith("*")`), both using `getCommentsBefore`.
- A tag-aware check (e.g. `/(^|\s)@default(\s|$)/m` on JSDoc comments whose value starts with `*`) plus a same-line/adjacency check would address all three.
